src/lib/reasoners/bitv.ml

```diff
@@ -785,35 +799,40 @@ module Shostak(X : ALIEN) = struct
             eq, csub, if c then Some pat else None
         end
       | st :: eq, n :: pt when st.sz < n ->
+        assert (not (Lists.is_empty (fst subs)));
+        assert (match st.bv with S_Var { sorte = B; _ } -> true | _ -> false);
```

Comment:
The `assert` should go after the explanatory comment rather than before, since the comment explains the rationale for the asserts.

Reply:
Done.